src/coreclr/jit/flowgraph.cpp

```diff
@@ -722,6 +722,9 @@ GenTreeCall* Compiler::fgGetStaticsCCtorHelper(CORINFO_CLASS_HANDLE cls, CorInfo
             // type = TYP_BYREF;
             break;
 
+        case CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED:
+            FALLTHROUGH;
```

Comment:
is `FALLTHROUGH` needed here?

Reply:
Yes, because we want `bNeedClassID = false`, `callFlags |= GTF_CALL_HOISTABLE` and `type = TYP_I_IMPL`.

Reply:
why it can't be just
```c
case CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED:
case CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE_NOCTOR:
```
?

Reply:
yes, I think I was tracking `typeIndex` at one point, but removed it. No `FALLTHROUGH` needed here.